feat(serve): use enigma mocker when rendering fixture (#710)
qlik-oss/nebula.js: commands/serve/docs/fixture-format.md

# Fixture format

## Purpose

The fixture is a set of configurations which specifies the visualization to render and how it behaves. For example, type of visualization, what data to display, theme to use and which feature flags to activate.

Rendering a visualization from a fixture multiple times yields the same result. This makes fixture rendering ideal to use in tests to verify that the visualization works as expected.

## Example

```js
import gridChart from '@nebula.js/sn-grid-chart';

export default () => {
  return {
    type: 'sn-grid-chart',
    load: async () => gridChart,
    instanceConfig: {
      context: {
        theme: 'dark',
      }
    },
    snConfig: {
      ...
    },
    genericObjects: [ ... ]
  };
}
```

## Fixture configurations

### `type`

Name of visualization. For example, `sn-grid-chart`.

### `load`

Function loading visualization to render.

Example using imports:

```js
import simpleVisualization from './simpleVisualization';
export default () => ({
  load: async () => gridChart,
  ...
});

> 

Reviewer: ```suggestion
```

```

> 

Reviewer: Something is a little wrong with this code snippet.. Don't thing the code block is properly closed here.

# Fixture format

## Purpose

> The fixture is a set of configurations which specifies the visualization to render and how it behaves. For example, title of the visuzalition, what data to display, theme to use and which feature flags to activate.

Reviewer: ```suggestion
The fixture is a set of configurations which specifies the visualization to render and how it behaves. For example, type of visualization, what data to display, theme to use and which feature flags to activate.
```

> Rendering a visualization from a fixture multiple times yields the same result. This makes fixture rendering ideal to use in tests to verify that the visualziation works as expected.

Reviewer: ```suggestion
Rendering a visualization from a fixture multiple times yields the same result. This makes fixture rendering ideal to use in tests to verify that the visualization works as expected.
```

> Rendering a visuzaliation from a fixture multiple times yields the same result. This makes fixture rendering ideal to use in tests to verify that no the visualziation works as expected.

Reviewer 2: Remove `no` from `to use in tests to verify that no the visualziation works`.

## Example

```js

> import gridChart from '@nebula.js/grid-chart';

Reviewer: ```suggestion
import gridChart from '@nebula.js/sn-grid-chart';
```

export default () => {
  return {
    type: 'sn-grid-chart',
    load: async () => gridChart,
    instanceConfig: {
      context: {
        theme: 'dark',
      }
    },
    snConfig: {
      ...
    },
    genericObjects: [ ... ]
  };
}
```

## Fixture configurations

### `type`

Name of visualization. For example, `sn-grid-chart`.

### `load`

Function loading visualization to render.

Example using imports:

```js
import gridChart from '@nebula.js/grid-chart';
export default () => ({
  load: async () => gridChart,
  ...
});
```

Example:

```js

> const visuzalition = (env) => ({

Reviewer 2: `visuzalition`

> ```js
> import gridChart from '@nebula.js/grid-chart';
> export default () => ({
>   load: async () => gridChart,
>   ...
> });
> ```
> 
> Example:
> 
> ```js
> const visualization = (env) => ({
>   qae: {
>     properties: {
>       qHyperCubeDef: {},
>       simpleMath: {
>         qValueExpression: {
>           qExpr: '1+1'
>         }
>       }
>     },
>   },
>   component() {
>     const layout = useLayout();
>     console.log(layout); // { qHyperCube: , simpleMath: 2 }
>   },
> });
> 
> export default () => ({
>   load: async () => visualization,
>   ...
> });
> ```

Reviewer: ```suggestion
```js
import simpleVisualization from './simpleVisualization';
export default () => ({
  load: async () => gridChart,
  ...
});
```

Simple supernova example:

```js
const simpleVisualization = () => ({
  qae: {
    // ...
  },
  component() {
    const layout = useLayout();
    console.log(layout);
  },
});

export default simpleVisualization;
```
```

### `instanceConfig`

> Configurations when initating embed instance (`embed(app, instanceConfig)`).

Reviewer 2: `initating`

```js
export default () => ({
  context: {
    theme: 'dark'
  },
  ...
});
```

```js
export default () => ({
  context: {
    constraints: {
      select: true
    },
    flags: {
      IMPROVED_GRID_CHART: true
    }
  },
  ...
});
```

### `snConfig`

> Configurations when rendering supernova visualization (`nebbie.render({ snConfig })`).
> 
> ```js
> export default () => ({
>   snConfig: {
>     examples: {
>       ...
>     }
>   },
>   ...
> });
> ```

Reviewer: ```suggestion
Configurations when rendering supernova visualization (`nebbie.render(snConfig)`).

```js
export default () => ({
  snConfig: {
    options: {
      myOption: 'option'
    },
    // ...
  },
});
```
```
